osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/EncryptionDelegate.java

```diff
@@ -8,7 +8,12 @@
  */
 package org.opensmartgridplatform.shared.security;
 
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
+
 public interface EncryptionDelegate {
-    EncryptedSecret encrypt(EncryptionProviderType encryptionProviderType, Secret secret, String keyReference);
-    Secret decrypt(EncryptedSecret secret, String keyReference);
+    EncryptedSecret encrypt(EncryptionProviderType encryptionProviderType, byte[] secret, String keyReference)
+            throws EncrypterException;
+    byte[] decrypt(EncryptedSecret secret, String keyReference) throws EncrypterException;
+    byte[] generateAes128BitsSecret(EncryptionProviderType encryptionProviderType, String keyReference)
+            throws EncrypterException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)